src/App.tsx

```diff
@@ -17,40 +45,33 @@ export const App: React.FC = () => {
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleQuery}
+            onClick={() => setHasClick(query.length > 0)}
```

Comment:
it's better to use "focus" and "blur" events to handle it
current implementation doesn't work as expected